test/unit/spriteManipulation.js

```diff
@@ -393,3 +393,68 @@ test('setPropRandom set sprite y properties between 50 and 350', async t => {
 
   nativeAPI.reset();
 });
+
+test('prev, next, and rand dance move will throw when not enough dance moves', async t => {
+
+  const subTest = async ({moveCount = 1, testCode}) => {
+    const nativeAPI = await helpers.createDanceAPI();
+    nativeAPI.play({
+      bpm: 120,
+    });
+
+    // Mock cat animation poses
+    for (let i = 0; i < moveCount; i++) {
+      nativeAPI.setAnimationSpriteSheet("CAT", i, {}, () => {});
+      nativeAPI.world.MOVE_NAMES.push({
+        name: `move${i}`,
+        rest: i === 0,
+      });
+    }
+    nativeAPI.world.fullLengthMoveCount = moveCount;
+    nativeAPI.world.restMoveCount = 1;
+
+    const sprite = nativeAPI.makeNewDanceSprite("CAT", null, {x: 200, y: 200});
+
+    testCode({ nativeAPI, sprite });
+
+    nativeAPI.reset();
+  };
+
+  // Verify invalid number of tests:
+  await subTest({ testCode: ({nativeAPI, sprite}) => {
+    let error = null;
+    try {
+      // Requesting 'rand' when only one dance should fail
+      nativeAPI.getNewChangeMove(sprite, 'rand', 1);
+    } catch (e) {
+      error = e;
+    }
+    t.notEqual(error, null, "next/prev/rand requires that we have 2 or more non-resting full length moves");
```

Comment:
I think we could use `t.throws`
https://github.com/substack/tape#tthrowsfn-expected-msg

Reply:
Spent some time investigating this, but abandoned after a while to not get blocked by it.